### ISSUE TITLE
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds...

### ISSUE THREAD

**Issue by SudhindraAnegondhi** [flutter/flutter#59669]
I am developing a cross platform app to run on desktops and mobiles - hence I am using flutter master channel for my development. Since two days my linux implementation kept throwing clip board exceptions without any clipboard events.  So when I saw there was an upgrade available, I tried to upgrade by issuing `flutter upgrade`. The upgrade was 'running' for more than 5 minutes, which of course was unprecedented and then finally failed saying it could not resolve 'pub.dartlang.org'. I checked my internet connection and it was functioning normally.

I repeated the process with the same results.

```
:~/projects/spincent/admin$ flutter channel master

Switching to flutter channel 'master'...
git: From https://github.com/flutter/flutter
git:    fe15d1e79..37cb7b7b4  master     -> origin/master
git: Switched to a new branch 'master'
git: Branch 'master' set up to track remote branch 'master' from 'origin'.
Successfully switched to flutter channel 'master'.
To ensure that you're on the latest build from this channel, run 'flutter
upgrade'
:~/projects/spincent/admin$ flutter upgrade
Downloading Dart SDK from Flutter engine 965fbbed1776545ee681cb57f3fb1d0c2000bbcb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  224M  100  224M    0     0  4876k      0  0:00:47  0:00:47 --:--:-- 5179k
Building flutter tool...
Could not resolve URL "https://pub.dartlang.org".
Error: Unable to 'pub upgrade' flutter tool. Retrying in five seconds... (9 tries left)

```
## Steps to Reproduce

1. Remove flutter directory, if present. Change directory where you intend installing flutter. `` Download latest installation bundle.- [https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_1.17.3-stable.tar.xz](url) from [https://flutter.dev/docs/get-started/install/linux](url)

2. . Install the stable version ` tar xf ~/Downloads/flutter_linux_1.17.3-stable.tar.xz
`. 
3. Add flutter tool to your path if not already `export PATH="$PATH:`pwd`/flutter/bin"`

4. Pre-download development binaries. ` flutter precache
`.

4. Change to channel dev. `flutter channel dev`, upgrade flutter when asked. `flutter upgrade`.

5. Change to channel master. `flutter channel master`. Type in `flutter upgrade` when prompted.

## Logs

No logs, flutter doctor command also starts off building flutter tool with the same result.

```
```

```

**Comment by VladyslavBondarenko** [comment]
Hi @SudhindraAnegondhi 
do you mean `flutter upgrade` is working when you run it on dev, but doesn't after switching to master?

**Comment by SudhindraAnegondhi** [comment]
That is correct.

**Comment by iapicca** [comment]
Hi @SudhindraAnegondhi 
following the steps you provided on my linux machine

<details>
  <summary>STEPS 1-4</summary>

```bash
francesco@francesco-yoga720  ~/development  wget https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_1.17.3-stable.tar.xz                               ✔  1576  14:39:23 
--2020-06-17 14:39:48--  https://storage.googleapis.com/flutter_infra/releases/stable/linux/flutter_linux_1.17.3-stable.tar.xz
Resolving storage.googleapis.com (storage.googleapis.com)... 2a00:1450:400f:80c::2010, 216.58.207.240
Connecting to storage.googleapis.com (storage.googleapis.com)|2a00:1450:400f:80c::2010|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 475664360 (454M) [application/x-gtar]
Saving to: ‘flutter_linux_1.17.3-stable.tar.xz’

flutter_linux_1.17.3-stable.tar.xz                  100%[=================================================================================================================>] 453.63M  2.09MB/s    in 3m 9s   

2020-06-17 14:42:57 (2.40 MB/s) - ‘flutter_linux_1.17.3-stable.tar.xz’ saved [475664360/475664360]

 francesco@francesco-yoga720  ~/development  tar xf ~/development/flutter_linux_1.17.3-stable.tar.xz                                                                                  ✔  1577  14:42:57 
 francesco@francesco-yoga720  ~/development  which flutter                                                                                                                            ✔  1578  14:43:49 
/home/francesco/development/flutter/bin/flutter
 francesco@francesco-yoga720  ~/development  flutter precache                                                                                                                         ✔  1579  14:43:55 
Downloading Android Maven dependencies...                           0.7s

```                 

</details>

<details>
  <summary>STEP 5</summary>

```bash
 francesco@francesco-yoga720  ~/development  cd                                                                                                                                       ✔  1580  14:44:13 
 francesco@francesco-yoga720  ~  flutter channel dev                                                                                                                                  ✔  1581  14:45:14 
Switching to flutter channel 'dev'...
git: From https://github.com/flutter/flutter
git:  + 2738a1148...f994b7697 beta                   -> origin/beta  (forced update)
git:  * [new branch]          dds_integration_take_2 -> origin/dds_integration_take_2
git:    6135091de..d9653445f  dev                    -> origin/dev
git:    a436fe768..f994b7697  flutter-1.19-candidate.4 -> origin/flutter-1.19-candidate.4
git:  * [new branch]          flutter-1.19-candidate.5 -> origin/flutter-1.19-candidate.5
git:  * [new branch]          flutter-1.20-candidate.0 -> origin/flutter-1.20-candidate.0
git:    28138ec5e..37cb7b7b4  master                 -> origin/master
git:  * [new tag]             1.19.0-4.1.pre         -> 1.19.0-4.1.pre
git:  * [new tag]             1.19.0-5.0.pre         -> 1.19.0-5.0.pre
git:  * [new tag]             1.20.0-0.0.pre         -> 1.20.0-0.0.pre
git:  * [new tag]             1.19.0-4.0.pre         -> 1.19.0-4.0.pre
git: Switched to a new branch 'dev'
git: Branch 'dev' set up to track remote branch 'dev' from 'origin'.
Successfully switched to flutter channel 'dev'.
To ensure that you're on the latest build from this channel, run 'flutter upgrade'
 francesco@francesco-yoga720  ~  flutter upgrade                                                                                                                                      ✔  1582  14:45:26 
Downloading Dart SDK from Flutter engine e8c13aa012c915c6a290d2a85950af67303e75f9...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  224M  100  224M    0     0  2154k      0  0:01:46  0:01:46 --:--:-- 4159k
Building flutter tool...
Flutter is already up to date on channel dev
Flutter 1.20.0-0.0.pre • channel dev • https://github.com/flutter/flutter.git
Framework • revision d9653445f4 (8 days ago) • 2020-06-09 18:43:03 -0400
Engine • revision e8c13aa012
Tools • Dart 2.9.0 (build 2.9.0-14.0.dev 5c1376615e)

```                 

</details>

<details>
  <summary>STEP 6</summary>

```bash
 francesco@francesco-yoga720  ~  flutter channel master                                                                                                                               ✔  1583  14:47:59 
Downloading package sky_engine...                                   1.2s
Downloading flutter_patched_sdk tools...                            4.5s
Downloading flutter_patched_sdk_product tools...                    4.3s
Downloading linux-x64 tools...                                     12.5s
Downloading linux-x64/font-subset tools...                          1.8s
Switching to flutter channel 'master'...
git: Switched to a new branch 'master'
git: Branch 'master' set up to track remote branch 'master' from 'origin'.
Successfully switched to flutter channel 'master'.
To ensure that you're on the latest build from this channel, run 'flutter upgrade'
 francesco@francesco-yoga720  ~  flutter upgrade                                                                                                                                      ✔  1584  14:49:20 
Downloading Dart SDK from Flutter engine 965fbbed1776545ee681cb57f3fb1d0c2000bbcb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  224M  100  224M    0     0  2634k      0  0:01:27  0:01:27 --:--:-- 3045k
Building flutter tool...
Flutter is already up to date on channel master
Flutter 1.20.0-1.0.pre.83 • channel master • https://github.com/flutter/flutter.git
Framework • revision 37cb7b7b48 (3 hours ago) • 2020-06-17 14:28:01 +0530
Engine • revision 965fbbed17
Tools • Dart 2.9.0 (build 2.9.0-14.0.dev 2b917f5b6a)

```                 

</details>

<details>
  <summary>info</summary>

```console
   Static hostname: francesco-yoga720
         Icon name: computer-convertible
           Chassis: convertible
        Machine ID: ...
           Boot ID: ...
  Operating System: Ubuntu 18.04.4 LTS
            Kernel: Linux 5.3.0-59-generic
      Architecture: x86-64
```                 

</details>

I cannot reproduce the behavior you described
Thank you

**Comment by SudhindraAnegondhi** [comment]
Mystery deepens. Let me try again and let you know.

**Comment by SudhindraAnegondhi** [comment]
The problem persisted after removing dart from the system by `sudo apt remove dart` and then I rebooted and tried it worked. It's still a mystery. Thank you for your help.
```
:~$  git clone -b master https://github.com/flutter/flutter.git
Cloning into 'flutter'...
remote: Enumerating objects: 15, done.
remote: Counting objects: 100% (15/15), done.
remote: Compressing objects: 100% (15/15), done.
remote: Total 243427 (delta 2), reused 1 (delta 0), pack-reused 243412
Receiving objects: 100% (243427/243427), 102.57 MiB | 2.75 MiB/s, done.
Resolving deltas: 100% (186896/186896), done.
:~$ flutter --version
Downloading Dart SDK from Flutter engine 965fbbed1776545ee681cb57f3fb1d0c2000bbcb...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  224M  100  224M    0     0  4353k      0  0:00:52  0:00:52 --:--:-- 4356k
Building flutter tool...
Flutter 1.20.0-1.0.pre.83 • channel master •
https://github.com/flutter/flutter.git
Framework • revision 37cb7b7b48 (4 hours ago) • 2020-06-17 14:28:01 +0530
Engine • revision 965fbbed17
Tools • Dart 2.9.0 (build 2.9.0-14.0.dev 2b917f5b6a)

```

**Comment by SudhindraAnegondhi** [comment]
it's resolved for now :)

**Comment by SudhindraAnegondhi** [comment]
Okay!!! finally traced the culprit. It was a local intermittent DNS issue. Fixed it.  Recreated the scenario and the mystery is finally resolved! :) Thanks guys!
